### PR TITLE
Object attributes are now built dynamically, so we can show context-s…

### LINKED ISSUE #559: Armor stats showing up in skillmods list
Armor stats showing up in skillmods list

image

### LINKED ISSUE #560: Armor parts showing bad strings
image

### PR DESCRIPTION
**Pull request by madsboddum**
…pecific attributes

Additional benefits include being able to reorder them at will, introducing new ones etc

Generally: Everything that was an object attribute before is now a field on SWGObject, or one of the subclasses, that best fit for the field. It also helped get rid of a lot of the ugly string manipulation.

Fixes #560
Fixes #559
There's no issue for it, but weapons also display "Your modified Speed" and "Your modified DPS" using the new dynamic item attributes.

### CODE REVIEW

File: src/main/java/com/projectswg/holocore/resources/support/objects/swg/tangible/ArmorCategory.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.projectswg.holocore.resources.support.objects.swg.tangible
+
+data class ArmorCategory(val id: String, val attributeName: String, val requiredCommand: String) {
+	
+	companion object {
+		val assault = ArmorCategory("assault", "@obj_attr_n:armor_assault", "assault_move_mitigate_1")
+		val battle = ArmorCategory("battle", "@obj_attr_n:armor_battle", "battle_move_mitigate_1")
+		val reconnaissance = ArmorCategory("reconnaissance", "@obj_attr_n:armor_reconnaissance", "recon_move_mitigate_1");
+		
+		fun getById(id: String?): ArmorCategory? {
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/resources/support/objects/swg/tangible/ArmorCategory.kt:10]: I suggest using a `when` here

```kt
return when (id) {
    assault.id -> assault
    battle.id -> battle
    reconnaissance.id -> reconnaissance
    else -> null
}
```

Comment (madsboddum) [reply src/main/java/com/projectswg/holocore/resources/support/objects/swg/tangible/ArmorCategory.kt:10]: oh yeah, that's a good idea

File: src/main/java/com/projectswg/holocore/resources/support/objects/swg/tangible/LightsaberPowerCrystalQuality.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	companion object {
+		val poor = LightsaberPowerCrystalQuality("poor", "@jedi_spam:crystal_quality_0")
+		val fair = LightsaberPowerCrystalQuality("fair", "@jedi_spam:crystal_quality_1")
+		val good = LightsaberPowerCrystalQuality("good", "@jedi_spam:crystal_quality_2")
+		val quality = LightsaberPowerCrystalQuality("quality", "@jedi_spam:crystal_quality_3")
+		val select = LightsaberPowerCrystalQuality("select", "@jedi_spam:crystal_quality_4")
+		val premium = LightsaberPowerCrystalQuality("premium", "@jedi_spam:crystal_quality_5")
+		val flawless = LightsaberPowerCrystalQuality("flawless", "@jedi_spam:crystal_quality_6")
+
+		fun getById(id: String?): LightsaberPowerCrystalQuality? {
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/resources/support/objects/swg/tangible/LightsaberPowerCrystalQuality.kt:14]: Same `when` suggestion here - although this has enough items I think it's definitely worth changing haha

Comment (madsboddum) [reply src/main/java/com/projectswg/holocore/resources/support/objects/swg/tangible/LightsaberPowerCrystalQuality.kt:14]: agreed, changed that too

**Pull request merged**